File: frontend/src/types/Account.ts
import TypeMapper from "@/mapper/TypeMapper";

interface Account {
    id: number;
    label: string;
    iban: string | null;
    startBalance: number;
    active: boolean;
    deposits: boolean;
}

export interface AccountDTO {
    id: number;
    label: string;
    iban: string | null;
    startBalance: number;
    active: boolean;
    deposits: boolean;
}

export const accountMapper: TypeMapper<Account, AccountDTO> = {
    fromDTO: (dto: AccountDTO) => dto,
    toDTO: (model: Account) => model,
};

export default Account;

File: backend/src/main/java/com/lennartmoeller/finance/model/Account.java
package com.lennartmoeller.finance.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;

@Data
@Entity
@EqualsAndHashCode(of = "id", callSuper = false)
@RequiredArgsConstructor
@Table(name = "accounts")
public class Account extends BaseModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String label;

    @Column(unique = true)
    private String iban;

    @Column(nullable = false)
    private Long startBalance;

    @Column(nullable = false)
    private Boolean active = true;

    @Column(nullable = false)
    private Boolean deposits = false;
}
